src/queue/memoryQueue.ts

```diff
@@ -0,0 +1,65 @@
+import { QueueId, QueueInterface } from "./queue";
+import { DSNPMessage } from "../messages/messages";
+
+export const IdDoesNotExist = new Error("No message matching the given ID exists.");
+
+/**
+ * MemoryQueue provides a simple, in-memory store for DSNP messages to be
+ * announced. This adapter is the default if no other is provided in
+ * configuration.
+ */
+export default class MemoryQueue implements QueueInterface {
```

Comment:
Just want to double check that we don't want to make this a module instead of a class.

Reply:
I made it a module at first, but the class version is preferable for resetting the queue in memory. As a module, dumping the entire queue and starting over from empty is a pain because it requires re-importing.

Reply:
... I don't quite understand the response. Resetting the queue would need to happen in the same way for the class version no? You cannot just `setConfig({queue: new Queue()})` because after you committed something else might still be holding on to a reference of the old queue.